### PR TITLE
gUU and guu

### LINKED ISSUE #1423: gUU and guu doesn't work
Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----

### What did you do?

Pressed gUU or guu keys in normal mode. The same for gUgU and gugu.

### What did you expect to happen?

The whole line should be in uppercase (gUU) or in lowercase (guu).

guu

### What happened instead?

Nothing.

### Technical details:

* VSCode Version: 1.10.2
* VsCodeVim Version: 0.6.11
* OS: Windows 10 Pro

Thank you!

### PR DESCRIPTION
**Pull request by xconverge**
fixes #1423

### CODE REVIEW

File: src/actions/actions.ts [hunk 1]
```diff
@@ -2082,7 +2082,7 @@ export class UpperCaseOperator extends BaseOperator {
       await TextEditor.replace(range, text.toUpperCase());
 
       vimState.currentMode = ModeName.Normal;
-      vimState.cursorPosition = start;
+      vimState.cursorPosition = vimState.cursorPositionJustBeforeAnythingHappened[0];
```
Comment (johnfn) [inline src/actions/actions.ts]: Why is this necessary over using start? (Also the [0] won't be multi cursor friendly like start and end are)

File: src/actions/actions.ts [hunk 2]
```diff
@@ -5132,6 +5133,32 @@ class MoveFormat extends BaseMovement {
 }
 
 @RegisterAction
+class MoveUpercaseLine extends BaseMovement {
```
Comment (johnfn) [inline src/actions/actions.ts]: Can you leave a comment here saying that these are for guu and gUU (for searchability mostly)

File: src/actions/actions.ts (hunk 1 above)
Comment (xconverge) [inline src/actions/actions.ts]: this entire command is already not multicursor friendly though

start in the case of gUU is start of line and end is end of line

gUU is not supposed to move the cursor so it needs to perform the action and remain in place

Comment (johnfn) [inline src/actions/actions.ts]: gUU looks like it just moves the cursor to the beginning of the line or am I wrong?

Comment (xconverge) [inline src/actions/actions.ts]: you are right...I am so confused, I wonder if it does the same thing in gvim, I tried yesterday...I will change to start, thats easy

Comment (johnfn) [inline src/actions/actions.ts]: haha no worries

File: src/actions/actions.ts [hunk 3]
```diff
@@ -5131,6 +5132,34 @@ class MoveFormat extends BaseMovement {
   }
 }
 
+// Used for gUU
```
Comment (johnfn) [inline src/actions/actions.ts]: It'd probably be good to move these next to the other weirdo motions I made for dd yy cc etc.

Comment (xconverge) [inline src/actions/actions.ts]: they are....

Comment (johnfn) [inline src/actions/actions.ts]: oh, nice! i didn't hit that little fold out button enough.

Comment (xconverge) [inline src/actions/actions.ts]: yea I think the > and = are also of the same weirdo type

**Pull request merged**